src/main/scala/com/criteo/hive/SaslClient.scala

```diff
@@ -0,0 +1,73 @@
+package com.criteo.hive
+
+import collection.JavaConversions._
+import java.util.Base64
+import javax.security.auth.callback.{Callback, CallbackHandler, NameCallback, PasswordCallback}
+import javax.security.sasl.{RealmCallback}
+
+import org.apache.hadoop.security.SaslRpcServer.AuthMethod
+import org.apache.hadoop.security.UserGroupInformation
+import org.apache.hadoop.security.token.{Token, TokenIdentifier}
+import org.apache.thrift.transport.{TSaslClientTransport, TTransport}
+
+/**
+  * Logic for the SASL Kerberos client
+  */
+object SaslClient {
+
+  def encodeIdentifier(identifier: Array[Byte]): String =
+    new String(Base64.getEncoder.encode(identifier))
+
+  def encodePassword(password: Array[Byte]): Array[Char] =
+    new String(Base64.getEncoder.encode(password)).toCharArray
+
+  private class SaslClientCallbackHandler(val token: Token[_ <: TokenIdentifier]) extends CallbackHandler {
```

Comment:
The Type constraint on val token is maybe better placed as a generic parameter for the class, eg:
```
private class SaslClientCallbackHandler[T <: TokenIdentifier](val token: T)
```